src/3d/terrain/qgsterraintileloader_p.cpp

```diff
@@ -64,6 +64,11 @@ void QgsTerrainTileLoader::loadTexture()
   mTextureJobId = mTerrain->textureGenerator()->render( mExtentMapCrs, mNode->tileId(), mTileDebugText );
 }
 
+void QgsTerrainTileLoader::loadTextureSynchronously()
```

Comment:
As suggested above - I would like to avoid this synchronous mode altogether and have waitForFinished() instead which would handle that.